### PR TITLE
[KYUUBI #2084] Support arbitrary parameters for KyuubiConf

### LINKED ISSUE #2084: [Improvement] Support arbitrary parameters for KyuubiConf or kyuubi prefixed only
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-kyuubi/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

  private def loadFromMap(props: Map[String, String] = Utils.getSystemProperties): KyuubiConf = {
    for ((key, value) <- props if key.startsWith("kyuubi.") || key.startsWith("spark.") ||
        // for user specific defaults
        key.startsWith("___")) {
      set(key, value)
    }
    this
  }

In KyuubiConf, we now load kyuubi.* and spark.* from the kyuubi-defaults.conf config file or sys.props, but we now support other engine types, we shall support all engine conf or remove spark.* here

### How should we improve?

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by jiaoqingbo**
### _Why are the changes needed?_

fix #2084 

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: kyuubi-common/src/test/resources/kyuubi-defaults.conf [hunk 1]
```diff
@@ -21,4 +21,7 @@ spark.kyuubi.yes no
 
 spark.user.test a
 ___userb___.spark.user.test b
-___userc___.spark.user.test c
\ No newline at end of file
+___userc___.spark.user.test c
+
+
+flink.jobmanager.rpc.address localhost
```
Comment (yaooqinn) [inline kyuubi-common/src/test/resources/kyuubi-defaults.conf:27]: we can add more, hive hadoop java trino balabal

**Review by yaooqinn — APPROVED** [review]

**Comment by jiaoqingbo** [comment]
- get catalogs for flink sql *** FAILED ***
  java.sql.SQLException: Could not open client transport with JDBC Uri: jdbc:hive2://localhost:36445/;: Failed to detect the root cause, please check /home/runner/work/incubator-kyuubi/incubator-kyuubi/integration-tests/kyuubi-flink-it/target/work/runner/kyuubi-flink-sql-engine.log.0 at server side if necessary. The last 10 line(s) of log are:
Error: Could not find or load main class 64-Bittor-kyuubi/incubator-kyuubi/externals/kyuubi-flink-sql-engine doesn't match production directory, assuming in development environment...
  at org.apache.kyuubi.jdbc.hive.KyuubiConnection.<init>(KyuubiConnection.java:205)
  at org.apache.kyuubi.jdbc.KyuubiHiveDriver.connect(KyuubiHiveDriver.java:56)
  at java.sql.DriverManager.getConnection(DriverManager.java:664)
  at java.sql.DriverManager.getConnection(DriverManager.java:247)
  at org.apache.kyuubi.operation.JDBCTestHelper.$anonfun$withMultipleConnectionJdbcStatement$1(JDBCTestHelper.scala:56)
  at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:286)
  at scala.collection.IndexedSeqOptimized.foreach(IndexedSeqOptimized.scala:36)
  at scala.collection.IndexedSeqOptimized.foreach$(IndexedSeqOptimized.scala:33)
  at scala.collection.mutable.WrappedArray.foreach(WrappedArray.scala:38)
  at scala.collection.TraversableLike.map(TraversableLike.scala:286)
  ...
  Cause: java.sql.SQLException: Failed to detect the root cause, please check /home/runner/work/incubator-kyuubi/incubator-kyuubi/integration-tests/kyuubi-flink-it/target/work/runner/kyuubi-flink-sql-engine.log.0 at server side if necessary. The last 10 line(s) of log are:
Error: Could not find or load main class 64-Bittor-kyuubi/incubator-kyuubi/externals/kyuubi-flink-sql-engine doesn't match production directory, assuming in development environment...
  at org.apache.kyuubi.jdbc.hive.KyuubiConnection.waitLaunchEngineToComplete(KyuubiConnection.java:1680)
  at org.apache.kyuubi.jdbc.hive.KyuubiConnection.<init>(KyuubiConnection.java:180)
  at org.apache.kyuubi.jdbc.KyuubiHiveDriver.connect(KyuubiHiveDriver.java:56)
  at java.sql.DriverManager.getConnection(DriverManager.java:664)
  at java.sql.DriverManager.getConnection(DriverManager.java:247)
  at org.apache.kyuubi.operation.JDBCTestHelper.$anonfun$withMultipleConnectionJdbcStatement$1(JDBCTestHelper.scala:56)
  at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:286)
  at scala.collection.IndexedSeqOptimized.foreach(IndexedSeqOptimized.scala:36)
  at scala.collection.IndexedSeqOptimized.foreach$(IndexedSeqOptimized.scala:33)
  at scala.collection.mutable.WrappedArray.foreach(WrappedArray.scala:38)
  ...
- execute statement - select column name with dots *** FAILED ***
  java.sql.SQLException: Could not open client transport with JDBC Uri: jdbc:hive2://localhost:36445/;: Failed to detect the root cause, please check /home/runner/work/incubator-kyuubi/incubator-kyuubi/integration-tests/kyuubi-flink-it/target/work/runner/kyuubi-flink-sql-engine.log.1 at server side if necessary. The last 10 line(s) of log are:
Error: Could not find or load main class 64-Bittor-kyuubi/incubator-kyuubi/externals/kyuubi-flink-sql-engine doesn't match production directory, assuming in development environment...
  at org.apache.kyuubi.jdbc.hive.KyuubiConnection.<init>(KyuubiConnection.java:205)
  at org.apache.kyuubi.jdbc.KyuubiHiveDriver.connect(KyuubiHiveDriver.java:56)
  at java.sql.DriverManager.getConnection(DriverManager.java:664)
  at java.sql.DriverManager.getConnection(DriverManager.java:247)
  at org.apache.kyuubi.operation.JDBCTestHelper.$anonfun$withMultipleConnectionJdbcStatement$1(JDBCTestHelper.scala:56)
  at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:286)
  at scala.collection.IndexedSeqOptimized.foreach(IndexedSeqOptimized.scala:36)
  at scala.collection.IndexedSeqOptimized.foreach$(IndexedSeqOptimized.scala:33)
  at scala.collection.mutable.WrappedArray.foreach(WrappedArray.scala:38)
  at scala.collection.TraversableLike.map(TraversableLike.scala:286)
  ...
  Cause: java.sql.SQLException: Failed to detect the root cause, please check /home/runner/work/incubator-kyuubi/incubator-kyuubi/integration-tests/kyuubi-flink-it/target/work/runner/kyuubi-flink-sql-engine.log.1 at server side if necessary. The last 10 line(s) of log are:
Error: Could not find or load main class 64-Bittor-kyuubi/incubator-kyuubi/externals/kyuubi-flink-sql-engine doesn't match production directory, assuming in development environment...
  at org.apache.kyuubi.jdbc.hive.KyuubiConnection.waitLaunchEngineToComplete(KyuubiConnection.java:1680)
  at org.apache.kyuubi.jdbc.hive.KyuubiConnection.<init>(KyuubiConnection.java:180)
  at org.apache.kyuubi.jdbc.KyuubiHiveDriver.connect(KyuubiHiveDriver.java:56)
  at java.sql.DriverManager.getConnection(DriverManager.java:664)
  at java.sql.DriverManager.getConnection(DriverManager.java:247)
  at org.apache.kyuubi.operation.JDBCTestHelper.$anonfun$withMultipleConnectionJdbcStatement$1(JDBCTestHelper.scala:56)
  at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:286)
  at scala.collection.IndexedSeqOptimized.foreach(IndexedSeqOptimized.scala:36)
  at scala.collection.IndexedSeqOptimized.foreach$(IndexedSeqOptimized.scala:33)
  at scala.collection.mutable.WrappedArray.foreach(WrappedArray.scala:38)
  ...
Run completed in 12 seconds, 869 milliseconds.
Total number of tests run: 2
Suites: completed 2, aborted 0
Tests: succeeded 0, failed 2, canceled 0, ignored 0, pending 0
*** 2 TESTS FAILED ***
Error:  Failed to execute goal org.scalatest:scalatest-maven-plugin:2.0.2:test (test) on project kyuubi-flink-it_2.12: There are test failures -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.scalatest:scalatest-maven-plugin:2.0.2:test (test) on project kyuubi-flink-it_2.12: There are test failures
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:972)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.apache.maven.plugin.MojoFailureException: There are test failures
    at org.scalatest.tools.maven.TestMojo.execute (TestMojo.java:109)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:972)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Error:  
Error:  Re-run Maven using the -X switch to enable full debug logging.
Error:  
Error:  For more information about the errors and possible solutions, please read the following articles:
Error:  [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
Error:  
Error:  After correcting the problems, you can resume the build with the command
Error:    mvn <args> -rf :kyuubi-flink-it_2.12
Error: Process completed with exit code 1.

**Comment by jiaoqingbo** [comment]
> * get catalogs for flink sql *** FAILED ***
>   java.sql.SQLException: Could not open client transport with JDBC Uri: jdbc:hive2://localhost:36445/;: Failed to detect the root cause, please check /home/runner/work/incubator-kyuubi/incubator-kyuubi/integration-tests/kyuubi-flink-it/target/work/runner/kyuubi-flink-sql-engine.log.0 at server side if necessary. The last 10 line(s) of log are:
>   Error: Could not find or load main class 64-Bittor-kyuubi/incubator-kyuubi/externals/kyuubi-flink-sql-engine doesn't match production directory, assuming in development environment...
>   at org.apache.kyuubi.jdbc.hive.KyuubiConnection.(KyuubiConnection.java:205)
>   at org.apache.kyuubi.jdbc.KyuubiHiveDriver.connect(KyuubiHiveDriver.java:56)
>   at java.sql.DriverManager.getConnection(DriverManager.java:664)
>   at java.sql.DriverManager.getConnection(DriverManager.java:247)
>   at org.apache.kyuubi.operation.JDBCTestHelper.$anonfun$withMultipleConnectionJdbcStatement$1(JDBCTestHelper.scala:56)
>   at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:286)
>   at scala.collection.IndexedSeqOptimized.foreach(IndexedSeqOptimized.scala:36)
>   at scala.collection.IndexedSeqOptimized.foreach$(IndexedSeqOptimized.scala:33)
>   at scala.collection.mutable.WrappedArray.foreach(WrappedArray.scala:38)
>   at scala.collection.TraversableLike.map(TraversableLike.scala:286)
>   ...
>   Cause: java.sql.SQLException: Failed to detect the root cause, please check /home/runner/work/incubator-kyuubi/incubator-kyuubi/integration-tests/kyuubi-flink-it/target/work/runner/kyuubi-flink-sql-engine.log.0 at server side if necessary. The last 10 line(s) of log are:
>   Error: Could not find or load main class 64-Bittor-kyuubi/incubator-kyuubi/externals/kyuubi-flink-sql-engine doesn't match production directory, assuming in development environment...
>   at org.apache.kyuubi.jdbc.hive.KyuubiConnection.waitLaunchEngineToComplete(KyuubiConnection.java:1680)
>   at org.apache.kyuubi.jdbc.hive.KyuubiConnection.(KyuubiConnection.java:180)
>   at org.apache.kyuubi.jdbc.KyuubiHiveDriver.connect(KyuubiHiveDriver.java:56)
>   at java.sql.DriverManager.getConnection(DriverManager.java:664)
>   at java.sql.DriverManager.getConnection(DriverManager.java:247)
>   at org.apache.kyuubi.operation.JDBCTestHelper.$anonfun$withMultipleConnectionJdbcStatement$1(JDBCTestHelper.scala:56)
>   at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:286)
>   at scala.collection.IndexedSeqOptimized.foreach(IndexedSeqOptimized.scala:36)
>   at scala.collection.IndexedSeqOptimized.foreach$(IndexedSeqOptimized.scala:33)
>   at scala.collection.mutable.WrappedArray.foreach(WrappedArray.scala:38)
>   ...
> * execute statement - select column name with dots *** FAILED ***
>   java.sql.SQLException: Could not open client transport with JDBC Uri: jdbc:hive2://localhost:36445/;: Failed to detect the root cause, please check /home/runner/work/incubator-kyuubi/incubator-kyuubi/integration-tests/kyuubi-flink-it/target/work/runner/kyuubi-flink-sql-engine.log.1 at server side if necessary. The last 10 line(s) of log are:
>   Error: Could not find or load main class 64-Bittor-kyuubi/incubator-kyuubi/externals/kyuubi-flink-sql-engine doesn't match production directory, assuming in development environment...
>   at org.apache.kyuubi.jdbc.hive.KyuubiConnection.(KyuubiConnection.java:205)
>   at org.apache.kyuubi.jdbc.KyuubiHiveDriver.connect(KyuubiHiveDriver.java:56)
>   at java.sql.DriverManager.getConnection(DriverManager.java:664)
>   at java.sql.DriverManager.getConnection(DriverManager.java:247)
>   at org.apache.kyuubi.operation.JDBCTestHelper.$anonfun$withMultipleConnectionJdbcStatement$1(JDBCTestHelper.scala:56)
>   at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:286)
>   at scala.collection.IndexedSeqOptimized.foreach(IndexedSeqOptimized.scala:36)
>   at scala.collection.IndexedSeqOptimized.foreach$(IndexedSeqOptimized.scala:33)
>   at scala.collection.mutable.WrappedArray.foreach(WrappedArray.scala:38)
>   at scala.collection.TraversableLike.map(TraversableLike.scala:286)
>   ...
>   Cause: java.sql.SQLException: Failed to detect the root cause, please check /home/runner/work/incubator-kyuubi/incubator-kyuubi/integration-tests/kyuubi-flink-it/target/work/runner/kyuubi-flink-sql-engine.log.1 at server side if necessary. The last 10 line(s) of log are:
>   Error: Could not find or load main class 64-Bittor-kyuubi/incubator-kyuubi/externals/kyuubi-flink-sql-engine doesn't match production directory, assuming in development environment...
>   at org.apache.kyuubi.jdbc.hive.KyuubiConnection.waitLaunchEngineToComplete(KyuubiConnection.java:1680)
>   at org.apache.kyuubi.jdbc.hive.KyuubiConnection.(KyuubiConnection.java:180)
>   at org.apache.kyuubi.jdbc.KyuubiHiveDriver.connect(KyuubiHiveDriver.java:56)
>   at java.sql.DriverManager.getConnection(DriverManager.java:664)
>   at java.sql.DriverManager.getConnection(DriverManager.java:247)
>   at org.apache.kyuubi.operation.JDBCTestHelper.$anonfun$withMultipleConnectionJdbcStatement$1(JDBCTestHelper.scala:56)
>   at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:286)
>   at scala.collection.IndexedSeqOptimized.foreach(IndexedSeqOptimized.scala:36)
>   at scala.collection.IndexedSeqOptimized.foreach$(IndexedSeqOptimized.scala:33)
>   at scala.collection.mutable.WrappedArray.foreach(WrappedArray.scala:38)
>   ...
>   Run completed in 12 seconds, 869 milliseconds.
>   Total number of tests run: 2
>   Suites: completed 2, aborted 0
>   Tests: succeeded 0, failed 2, canceled 0, ignored 0, pending 0
>   *** 2 TESTS FAILED ***
>   Error:  Failed to execute goal org.scalatest:scalatest-maven-plugin:2.0.2:test (test) on project kyuubi-flink-it_2.12: There are test failures -> [Help 1]
>   org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.scalatest:scalatest-maven-plugin:2.0.2:test (test) on project kyuubi-flink-it_2.12: There are test failures
>   at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
>   at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
>   at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
>   at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
>   at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
>   at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
>   at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
>   at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
>   at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
>   at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
>   at org.apache.maven.cli.MavenCli.execute (MavenCli.java:972)
>   at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
>   at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
>   at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
>   at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
>   at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
>   at java.lang.reflect.Method.invoke (Method.java:498)
>   at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
>   at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
>   at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
>   at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
>   Caused by: org.apache.maven.plugin.MojoFailureException: There are test failures
>   at org.scalatest.tools.maven.TestMojo.execute (TestMojo.java:109)
>   at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
>   at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
>   at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
>   at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
>   at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
>   at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
>   at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
>   at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
>   at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
>   at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
>   at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
>   at org.apache.maven.cli.MavenCli.execute (MavenCli.java:972)
>   at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
>   at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
>   at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
>   at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
>   at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
>   at java.lang.reflect.Method.invoke (Method.java:498)
>   at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
>   at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
>   at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
>   at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
>   Error:
>   Error:  Re-run Maven using the -X switch to enable full debug logging.
>   Error:
>   Error:  For more information about the errors and possible solutions, please read the following articles:
>   Error:  [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
>   Error:
>   Error:  After correcting the problems, you can resume the build with the command
>   Error:    mvn  -rf :kyuubi-flink-it_2.12
>   Error: Process completed with exit code 1.

can anyone tell me why my pr could cause this error？

**Comment by yaooqinn** [comment]
can we check the root cause in /home/runner/work/incubator-kyuubi/incubator-kyuubi/integration-tests/kyuubi-flink-it/target/work/runner/kyuubi-flink-sql-engine.log.0

**Comment by yaooqinn** [comment]
also cc @SteNicholas @yanghua

**Comment by jiaoqingbo** [comment]
> can we check the root cause in /home/runner/work/incubator-kyuubi/incubator-kyuubi/integration-tests/kyuubi-flink-it/target/work/runner/kyuubi-flink-sql-engine.log.0

this is only one line：
Error: Could not find or load main class 64-Bittor-kyuubi/incubator-kyuubi/externals/kyuubi-flink-sql-engine doesn't match production directory, assuming in development environment...

**Comment by yaooqinn** [comment]
> > can we check the root cause in /home/runner/work/incubator-kyuubi/incubator-kyuubi/integration-tests/kyuubi-flink-it/target/work/runner/kyuubi-flink-sql-engine.log.0
> 
> this is only one line： Error: Could not find or load main class 64-Bittor-kyuubi/incubator-kyuubi/externals/kyuubi-flink-sql-engine doesn't match production directory, assuming in development environment...

07:45:35.863 KyuubiSessionManager-exec-pool: Thread-82 INFO EngineRef: Launching engine:
/home/runner/work/incubator-kyuubi/incubator-kyuubi/externals/kyuubi-flink-sql-engine/bin/flink-sql-engine.sh
07:45:35.870 KyuubiSessionManager-exec-pool: Thread-82 INFO ProcBuilder: Logging to /home/runner/work/incubator-kyuubi/incubator-kyuubi/integration-tests/kyuubi-flink-it/target/work/runner/kyuubi-flink-sql-engine.log.1

**Comment by yaooqinn** [comment]
lets add some debug message in /home/runner/work/incubator-kyuubi/incubator-kyuubi/externals/kyuubi-flink-sql-engine/bin/flink-sql-engine.sh?

**Comment by jiaoqingbo** [comment]
> kyuubi-flink-sql-engine.log.1

In kyuubi-flink-sql-engine.log.1 ，the same content as kyuubi-flink-sql-engine.log.0

**Comment by yaooqinn** [comment]
looks flink-sql-engine.sh works wrongly.

File: externals/kyuubi-flink-sql-engine/bin/flink-sql-engine.sh [hunk 2]
```diff
@@ -26,14 +26,22 @@ if [[ -z "$FLINK_HOME" || ! -d "$FLINK_HOME" ]]; then
   exit 1
 fi
 
+echo -e "\nFLINK_HOME $FLINK_HOME"
```
Comment (pan3793) [inline externals/kyuubi-flink-sql-engine/bin/flink-sql-engine.sh:29]: you can use `set -x` to debug shell

**Comment by SteNicholas** [comment]
@jiaoqingbo, could you please rebase the master and squash your commits?

**Comment by jiaoqingbo** [comment]
cc @yaooqinn @pan3793

**Review by ulysses-you — APPROVED** [review]

**Comment by ulysses-you** [comment]
thanks, merging to master

**Pull request closed without merging**